api/filters.go

```diff
@@ -931,63 +337,6 @@ func (api *FilterAPI) getFilter(ctx context.Context, filterID string) (*models.F
 	return nil, errNotFound
 }
 
-func (api *FilterAPI) getOutput(r *http.Request, filterID string) (*models.Filter, error) {
-
-	ctx := r.Context()
-	logData := log.Data{"filter_output_id": filterID}
-
-	output, err := api.dataStore.GetFilterOutput(filterID)
-	if err != nil {
-		log.Error(err, logData)
-		return nil, err
-	}
-
-	logData["filter_blueprint_id"] = output.Links.FilterBlueprint.ID
-
-	// Hide private download links if request is not authenticated
-	if r.Header.Get(common.DownloadServiceHeaderKey) != api.downloadServiceToken {
-
-		log.Info("a valid download service token has not been provided. hiding private links", logData)
-
-		if output.Downloads != nil {
-			if output.Downloads.CSV != nil {
-				output.Downloads.CSV.Private = ""
-			}
-			if output.Downloads.XLS != nil {
-				output.Downloads.XLS.Private = ""
-			}
-		}
-	} else {
-		log.Info("a valid download service token has been provided. not hiding private links", logData)
-	}
-
-	//only return the filter if it is for published data or via authenticated request
-	if output.Published != nil && *output.Published == models.Published || common.IsCallerPresent(ctx) {
-		return output, nil
-	}
-
-	log.Info("unauthenticated request to access unpublished filter output", logData)
-
-	filter, err := api.getFilter(ctx, output.Links.FilterBlueprint.ID)
-	if err != nil {
-		log.Error(errors.New("failed to retrieve filter blueprint"), logData)
-		return nil, errFilterOutputNotFound
-	}
-
-	//filter has been published since output was last requested, so update output and return
-	if filter.Published != nil && *filter.Published == models.Published {
-		output.Published = &models.Published
-		if err := api.dataStore.UpdateFilterOutput(output); err != nil {
-			log.Error(err, logData)
-			return nil, errFilterOutputNotFound
-		}
-
-		return output, nil
-	}
-
-	return nil, errFilterOutputNotFound
-}
-
```

Comment:
consider moving the error declarations to the files they're used in. alo the checkFilterOptions function beneath this - should this be in the dimensions/options files?

Reply:
The checkFilterOptions function is only used within this file - the name would imply otherwise 😄